app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,237 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.NewTask;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+
+    private TaskService taskService;
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+    }
+
+    @Nested
+    @DisplayName("getTasks 메소드는")
+    class Describe_getTasks {
+        List<Task> subject() {
+            return taskService.getTasks();
+        }
+
+        @Nested
+        @DisplayName("DB에 데이터가 없는데 조회를 한다면")
+        class Context_with_empty_db {
+            @Test
+            @DisplayName("빈 ArrayList를 리턴한다")
+            void it_returns_empty_arraylist() {
+                List<Task> findTasks = subject();
+
+                assertThat(findTasks).hasSize(0);
+                assertThat(findTasks).isEqualTo(new ArrayList<>());
+            }
+        }
+
+        @Nested
+        @DisplayName("DB에 데이터가 있는데 조회를 한다면")
+        class Context_with_non_empty_db extends NewTask {
+            private Task task1, task2;
+            private int expectedSize;
+
+            @BeforeEach
+            void prepareTests() {
+                task1 = withTitle("할 일 추가1");
+                task2 = withTitle("할 일 추가2");
+
+                taskService.createTask(task1);
+                taskService.createTask(task2);
+
+                expectedSize = taskService.getTasks().size();
+            }
+
+            @Test
+            @DisplayName("등록된 모든 할 일을 리턴한다")
+            void it_returns_all_tasks() {
+                // when
+                List<Task> findTasks = subject();
+
+                final int actualSize = findTasks.size();
+
+                // then
+                assertThat(actualSize).isEqualTo(expectedSize);
+                assertThat(findTasks.get(0).getTitle()).isEqualTo(task1.getTitle());
+                assertThat(findTasks.get(1).getTitle()).isEqualTo(task2.getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_createTask {
+        @Nested
+        @DisplayName("새로운 할 일을 추가한다면")
+        class Context_with_new_task extends NewTask {
```

Comment:
설명을 이어 붙여서 읽어보는 연습을 해보세요. 다음 두 문장을 비교해보면서 생각해 보세요.

- createTask 메소드는, 새로운 할 일을 추가한다면, DB에 등록 후 등록된 할 일을 리턴한다.
- createTask 메소드는, 할 일 정보가 주어지면, DB에 등록하고 등록된 할 일을 리턴한다.

```suggestion
    @Nested
    @DisplayName("createTask 메소드는")
    class Describe_createTask {
        @Nested
        @DisplayName("할 일이 주어지면")
        class Context_with_new_task extends NewTask {
```